### ISSUE TITLE
Layout/MultilineMethodCallIndentation: false positive & broken autocorrect for a multi-line chained argument inside a hash pair value

### ISSUE THREAD

**Issue by tim-semba** [rubocop/rubocop#15423]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False positive

### Cop name

Layout/MultilineMethodCallIndentation

### Steps to reproduce the problem

With `EnforcedStyle: indented`:

```yaml
# .rubocop.yml
Layout/MultilineMethodCallIndentation:
  EnforcedStyle: indented
```

```ruby
foo(
  key => Model.joins(
    Other
      .arel_table
      .join_sources
  )
)
```

The value of the hash pair (`key => Model.joins(...)`) is a method chain, and one of
its calls (`.joins`) receives a **multi-line chained argument** whose receiver
(`Other`) sits on its own line.

### Expected behavior

No offense. The inner chain (`.arel_table`, `.join_sources`) is correctly indented 2
spaces relative to **its own receiver** `Other`. This is what 1.84.0 and 1.85.x report
(no offense).

### Actual behavior

An offense is reported, and the inner chain is anchored to the **hash pair key line**
(`key =>`, column 2) instead of to its receiver `Other`:

```
example.rb:4:7: C: [Correctable] Layout/MultilineMethodCallIndentation: Use 2 (not 4) spaces for indenting an expression spanning multiple lines.
      .arel_table
      ^^^^^^^^^^^
example.rb:5:7: C: [Correctable] Layout/MultilineMethodCallIndentation: Use 2 (not 4) spaces for indenting an expression spanning multiple lines.
      .join_sources
      ^^^^^^^^^^^^^
```

Autocorrect (`-a`) then indents the chain to the **same column as (effectively left of)
its own receiver**, which is clearly wrong:

```ruby
foo(
  key => Model.joins(
    Other
    .arel_table
    .join_sources
  )
)
```

### RuboCop configuration

```yaml
Layout/MultilineMethodCallIndentation:
  EnforcedStyle: indented
```

### RuboCop version

```shell
1.88.0 (using Parser 3.3.11.1, Prism 1.9.0, rubocop-ast 1.49.1, analyzing as Ruby 3.4, running on ruby 3.4.9) [x86_64-linux]
```
